src/java/org/apache/cassandra/cql3/statements/schema/CreateIndexStatement.java

```diff
@@ -134,7 +134,7 @@ public Keyspaces apply(Keyspaces schema)
         if (table.isView())
             throw ire("Secondary indexes on materialized views aren't supported");
 
-        if (Keyspace.open(table.keyspace).getReplicationStrategy().hasTransientReplicas())
+        if (keyspace.createReplicationStrategy().hasTransientReplicas())
```

Comment:
I think it's permissible in slow paths like create index and other schema manipulations...